rs/bitcoin/ckbtc/minter/src/metrics.rs

```diff
@@ -1,9 +1,62 @@
 use crate::state;
-use std::cell::Cell;
+use std::cell::{Cell, RefCell};
+use std::collections::BTreeMap;
+use std::io::Error;
+use std::time::Duration;
 
 thread_local! {
     pub static GET_UTXOS_CLIENT_CALLS: Cell<u64> = Cell::default();
     pub static GET_UTXOS_MINTER_CALLS: Cell<u64> = Cell::default();
+    pub static UPDATE_CALL_LATENCY: RefCell<BTreeMap<usize,LatencyHistogram>> = RefCell::default();
+}
+
+pub(crate) const BUCKETS_MS: [u64; 7] = [500, 1_000, 2_000, 4_000, 8_000, 16_000, 32_000];
+
+#[derive(Default, Clone, Copy)]
+pub struct LatencyHistogram {
+    latency_buckets: [u64; BUCKETS_MS.len() + 1],
+    latency_sum: u64,
+}
+
+impl LatencyHistogram {
+    pub fn observe_latency(&mut self, latency: Duration) {
+        let latency_ms = latency.as_millis() as u64;
+        let bucket_index = BUCKETS_MS
+            .iter()
+            .enumerate()
+            .filter(|(_, bucket_upper_bound)| latency_ms <= **bucket_upper_bound)
```

Comment:
nit: use `find` would be equivalent to `filter().map().next()`.

Reply:
Sorry, should be `find_map`.

Reply:
Thank you for the tip!